src/test_utils.rs

```diff
@@ -43,21 +43,35 @@ pub fn get_lines_content<P: AsRef<Path>>(path: P) -> BTreeSet<String> {
     set
 }
 
+fn get_files_to_compare<P>(output_dir: &P, files_to_check: Option<Vec<&str>>) -> Vec<String>
+where
+    P: AsRef<Path>,
+{
+    match files_to_check {
+        None => fs::read_dir(output_dir.as_ref())
+            .unwrap()
+            .map(|f| f.unwrap().file_name().into_string().unwrap())
+            .collect(),
+        Some(v) => v.iter().map(|&f| f.to_string()).collect(),
+    }
+}
+
 pub fn compare_output_dir_with_expected<P: AsRef<Path>>(
```

Comment:
This method was particularly designed for CSV comparison (load all the lines in a `BTreeSet` and compare). But it doesn't work so well for XML comparison. So I created 2 new functions `compare_output_dir_with_expected_csv` and `compare_output_dir_with_expected_xml` with different behaviors.

In order to avoid change all-over the code, I kept `compare_output_dir_with_expected` as a call to the new function `compare_output_dir_with_expected_csv`. I will probably do a second pass in another PR to do the replacement everywhere and remove this `compare_output_dir_with_expected` function entirely.